### PR TITLE
Add `dotnet package search` documentation

### PR DESCRIPTION
**Pull request by Nigusu-Allehu**
## Summary

Add `dotnet package search` documentation.
Implementation: https://github.com/NuGet/NuGet.Client/pull/5466

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/tools/dotnet-package-search.md](https://github.com/dotnet/docs/blob/e56d055db72724d30dc2d12712bb23a6753aae9b/docs/core/tools/dotnet-package-search.md) | [dotnet package search](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-package-search?branch=pr-en-us-37747) |
| [docs/core/tools/index.md](https://github.com/dotnet/docs/blob/e56d055db72724d30dc2d12712bb23a6753aae9b/docs/core/tools/index.md) | [.NET CLI](https://review.learn.microsoft.com/en-us/dotnet/core/tools/index?branch=pr-en-us-37747) |

### CODE REVIEW

File: docs/core/tools/dotnet-package-search.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    The number of results to return. The default value is 20.
+
+- **`--skip`**
+
+    The number of results to skip, for pagination. The default value is 0.
+
+[!INCLUDE [help](../../../includes/cli-help.md)]
+
+## Examples
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:67]: See [dotnet tool search](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-search) for an example of how to format these examples and how to write the text that introduces each one. Also, note that the opening triple backtick for each output example can specify output as the type:
~~~
```output
~~~

File: docs/core/tools/dotnet-package-search.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+The `dotnet package search` command searches for a NuGet package.
+
+## Arguments
+
+- **`search terms`**
+
+  Specifies the search term to filter results. Use this argument to search for packages matching the provided query. Example: `dotnet package search json`.
+
+## Options
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:39]: We generally try to put the options in alphabetical order.

File: docs/core/tools/dotnet-package-search.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+## Synopsis
+
+```dotnetcli
+dotnet package search [search terms] [options]
+    [--source <SOURCE>]
+    [--exact-match]
+    [--prerelease]
+    [--interactive]
+    [--take <TAKE>]
+    [--skip <SKIP>]
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:24]: See [dotnet tool search](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-search) for an example of how to format the Synopsis section. Also we generally try to put the options in alphabetical order.

**Review by tdykstra — CHANGES_REQUESTED** [review]
Thanks for doing this, @Nigusu-Allehu! The content is great, but I noted some ways in which it could more closely follow the patterns we typically use for formatting the CLI docs.

**Comment by Nigusu-Allehu** [comment]
> Thanks for doing this, @Nigusu-Allehu! The content is great, but I noted some ways in which it could more closely follow the patterns we typically use for formatting the CLI docs.

Thank you for reviewing! I will revise this as soon as the product is complete and is about to be released.

File: docs/core/tools/dotnet-package-search.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+- **`--verbosity`**
+
+    Display this amount of details in the output: `normal`, `minimal`,`detailed`. The default is`normal`.
+
+[!INCLUDE [help](../../../includes/cli-help.md)]
+
+## Examples
+
+- Search NuGet.org for packages using the search term "Newtonsoft.Json"
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:80]:
```suggestion
- Search NuGet.org for packages that match the search term "Newtonsoft.Json," and render the output as a table with up to 20 packages:
```

File: docs/core/tools/dotnet-package-search.md [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+    ```dotnetcli
+    dotnet package search Newtonsoft.Json --source https://api.nuget.org/v3/index.json
+    ```
+
+    output a list of 20 packages in the source matching the search term.
+
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:87]:
```suggestion
```

File: docs/core/tools/dotnet-package-search.md [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        | Newtonsoft.Json.Bson                        | 1.0.2          |        | 554,641,545     |
+        |---------------------------------------------|----------------|--------|-----------------|
+        | Newtonsoft.Json.Schema                      | 3.0.15         |        | 39,648,430      |
+        |---------------------------------------------|----------------|--------|-----------------|
+        | Microsoft.AspNetCore.Mvc.NewtonsoftJson     | 7.0.12         |        | 317,067,823     |
+        |---------------------------------------------|----------------|--------|-----------------|
+        ...
+    ```
+
+- Search NuGet.org for packages using the search term "Newtonsoft.Json" and render the output as a json.
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:103]:
```suggestion
- Search NuGet.org for packages that match the search term "Newtonsoft.Json," and render the output as json.
```

File: docs/core/tools/dotnet-package-search.md [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+                "latestVersion": "7.0.0",
+                "totalDownloads": 12583471,
+                "owners": "dotnetfoundation, reactiveui"
+                },
+                {
+                "id": "Microsoft.Azure.Core.NewtonsoftJson",
+                "latestVersion": "2.0.0",
+                "totalDownloads": 6593746,
+                "owners": "azure-sdk, Microsoft"
+                },
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:164]:
```suggestion
```

File: docs/core/tools/dotnet-package-search.md [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+                "latestVersion": "8.0.3",
+                "totalDownloads": 17044287,
+                "owners": "aspnet, dotnetframework, Microsoft"
+                },
+                {
+                "id": "MassTransit.Newtonsoft",
+                "latestVersion": "8.1.3",
+                "totalDownloads": 5185238,
+                "owners": "phatboyg"
+                },
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:200]:
```suggestion
```

File: docs/core/tools/dotnet-package-search.md [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+                "latestVersion": "24.2.45748",
+                "totalDownloads": 3855727,
+                "owners": "Curiosity"
+                },
+                {
+                "id": "Newtonsoft.Json.FSharp",
+                "latestVersion": "3.2.2",
+                "totalDownloads": 245301,
+                "owners": "henrik feldt"
+                },
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:230]:
```suggestion
```

File: docs/core/tools/dotnet-package-search.md [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+                "totalDownloads": 113101,
+                "owners": "simoncropp"
+                }
+            ]
+            }
+        ]
+        }
+    ```
+
+- Search NuGet.org for packages using the search term "Newtonsoft.Json", show only two results and skip the first packages in the search result
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:243]:
```suggestion
- Search NuGet.org for packages using the search term "Newtonsoft.Json," show only two results, and skip the first packages in the search result:
```

File: docs/core/tools/dotnet-package-search.md [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+
+    ```dotnetcli
+    dotnet package search Newtonsoft.Json --source https://api.nuget.org/v3/index.json --skip 1 --take 2
+    ```
+
+    output a list of 2 packages after skipping 1 in the source matching the search term.
+
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:250]:
```suggestion
```

File: docs/core/tools/dotnet-package-search.md [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+    ```output
+        Source: https://api.nuget.org/v3/index.json
+        | Package ID                                  | Latest Version | Owners | Downloads       |
+        |---------------------------------------------|----------------|--------|-----------------|
+        | Newtonsoft.Json.Bson                        | 1.0.2          |        | 554,641,545     |
+        | Newtonsoft.Json.Schema                      | 3.0.15         |        | 39,648,430      |
+    ```
+
+- Search for packages that exactly match "Newtonsoft.Json" and list all available versions of it, ignoring any packages that contain "Newtonsoft.Json" as a part of their name or description but don't match it exactly.
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:259]:
```suggestion
- Search for packages that exactly match "Newtonsoft.Json" and list all available versions of it, ignoring any packages that contain "Newtonsoft.Json" as a part of their name or description but don't match it exactly:
```

File: docs/core/tools/dotnet-package-search.md [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+
+    ```dotnetcli
+    dotnet package search Newtonsoft.Json --source https://api.nuget.org/v3/index.json --exact-match
+    ```
+
+    output only an exact match.
+
```
Comment (tdykstra) [inline docs/core/tools/dotnet-package-search.md:266]:
```suggestion
```

File: docs/core/tools/index.md [hunk 14]
```diff
@@ -59,6 +59,7 @@ The following commands are installed by default:
 - [`nuget verify`](dotnet-nuget-verify.md) (Available since .NET 5 SDK)
 - [`nuget trust`](dotnet-nuget-trust.md) (Available since .NET 5 SDK)
 - [`nuget sign`](dotnet-nuget-sign.md) (Available since .NET 6 SDK)
+- [`package search`](dotnet-package-search.md) (Available since .NET 8.02 SDK)
```
Comment (tdykstra) [inline docs/core/tools/index.md:62]:
```suggestion
- [`package search`](dotnet-package-search.md) (Available since .NET 8.02xx SDK)
```

**Review by tdykstra — COMMENTED** [review]
Looks good, I left some suggestions.

**Review by tdykstra — APPROVED** [review]

**Comment by tdykstra** [comment]
@Nigusu-Allehu is this ready to merge?

**Review by donnie-msft — APPROVED** [review]

**Comment by Nigusu-Allehu** [comment]
> @Nigusu-Allehu Nigusu Yenework FTE is this ready to merge?

Yes

**Pull request merged**